### PR TITLE
[CI] activating MPI tests

### PR DESCRIPTION
**Pull request by philbucher**
This activates the MPI-tests in the CI. The prerequisites were figured out in different PRs and are already merged (see #6996)

Note that this is done "manual" at the moment.
The reason is that I want to first collect some experiences with the tests that I am familiar with, after some time this should be replaced by sth similar to the `run_tests.py` script

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -72,6 +72,118 @@ jobs:
         export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}/libs
         python3 kratos/python_scripts/run_tests.py -l small -c python3
 
+    - name: Running MPICore C++ tests (2 Cores)
+      run: |
+        . /opt/intel/mkl/bin/mklvars.sh
+        export PYTHONPATH=${PYTHONPATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}
+        export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}/libs
+        # mpiexec -np 2 python3 kratos/python_scripts/run_cpp_mpi_tests.py --using-mpi
```
Comment (sunethwarna) [inline .github/workflows/ci.yml:80]: Are these commented out purposely for testing?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -72,6 +72,118 @@ jobs:
...
+
+    - name: Running MPICore C++ tests (3 Cores)
+      run: |
+        . /opt/intel/mkl/bin/mklvars.sh
+        export PYTHONPATH=${PYTHONPATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}
+        export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}/libs
+        # mpiexec --oversubscribe -np 3 python3 kratos/python_scripts/run_cpp_mpi_tests.py --using-mpi
```
Comment (sunethwarna) [inline .github/workflows/ci.yml:87]: here also

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -72,6 +72,118 @@ jobs:
...
+
+    - name: Running MPICore C++ tests (4 Cores)
+      run: |
+        . /opt/intel/mkl/bin/mklvars.sh
+        export PYTHONPATH=${PYTHONPATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}
+        export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}/libs
+        # mpiexec --oversubscribe -np 4 python3 kratos/python_scripts/run_cpp_mpi_tests.py --using-mpi
```
Comment (sunethwarna) [inline .github/workflows/ci.yml:94]: here also

**Review by sunethwarna — COMMENTED** [review]
Minor comments only. Thanks, this is good for everyone :). I'll leave the approval to technical commitee.

**Comment by philbucher** [comment]
@sunethwarna the C++ MPI tests are currently failing: https://github.com/KratosMultiphysics/Kratos/issues/6998

Once fixed we will activate them

**Review by sunethwarna — APPROVED** [review]
Nice :)

**Comment by philbucher** [comment]
merging to see how it behaves in action

will refactor it in a bit once we collected a bit of experience

**Pull request merged**